coeus-impl/src/main/java/org/kuali/coeus/common/budget/impl/period/BudgetPeriodRule.java

```diff
@@ -340,6 +340,10 @@ private boolean isValidNewBudgetPeriod(Budget budget, BudgetPeriod newBudgetPeri
                         newBudgetPeriod.setBudgetPeriod(periodNum+1);
                     }
                     break;
+                }else if(newPeriodStartDate.compareTo(periodEndDate) <= 0) {
+                    dateCompareValue = "ERROR_NEW_PERIOD_START_BEFORE_PREVIOUS_END";
+                    saveErrors(dateCompareValue, errorMap, newPeriodDateParams);
+                    validNewBudgetPeriod = false;
```

Comment:
Ok....reading the code again I think you are correct. I was missing the part where the previous if would catch the second period and your case wouldn't. Although I also think you should put a break in there if you detect an error to avoid it looping further if there is a problem.